### PR TITLE
[billing] place a hold on cards before subscribing

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Additional credit card verification by placing a hold on the provided card before subscribing.

Screenshot 2023-03-10 at 19 57 36

## Related Issue(s)

Fixes #

## How to test

- use credit card: 4000000000000341 and see it failing
- use credit card: 4242424242424242 and see it succeeding

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/ee/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -2366,6 +2367,9 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
             return costCenter?.spendingLimit;
         } catch (error) {
             log.error(`Failed to subscribe '${attributionId}' to Stripe`, error);
+            if (error instanceof ClientError) {
+                throw new ResponseError(error.code, error.details);
```
Comment (geropl) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:2371]: This looks like we're starting to spill system internals to clients (again). What's the reason for doing so? Only for debuggin/testing? :thinking:

Comment (svenefftinge) [reply components/server/ee/src/workspace/gitpod-server-impl.ts:2371]: It's a user error to provide a non-working credit card the error message from stripe is telling the user what happened i.e. why their credit card doesn't work.

**Comment by geropl** [comment]
@svenefftinge Any idea how to get a working preview environment?

**Comment by geropl** [comment]
Ah, missed that we swit5ched to GHA :see_no_evil: RE-running that one...

**Comment by geropl** [comment]
Trying to rebase this one now

**Review by geropl — APPROVED** [review]
I could successfully test this, and the code looks good as well.

[This discussion](https://github.com/gitpod-io/gitpod/pull/16807/files#r1134259415) is not as much about this specific place, but more a general one so should never block this PR.

**Pull request merged**